cuqi/distribution/_distribution.py

```diff
@@ -84,42 +84,69 @@ def __init__(self, name=None, geometry=None, is_symmetric=None):
 
         is_symmetric : bool, default None
             Indicator if distribution is symmetric.
+
+        dim : int or None
+            Dimension of distribution. Cannot be passed together with geometry.
                         
         """
         super().__init__(name=name)
         self.is_symmetric = is_symmetric
-        self.geometry = geometry
+        if dim and geometry:
+            raise ValueError(f"Cannot specify both 'dim' and 'geometry'.")
+        if dim:
+            self.geometry = dim
+        else:
+            self.geometry = geometry
 
     @property
     def dim(self):
-        """ Return the dimension of the distribution.
-        
-        The dimension is automatically inferred from the mutable variables of the distribution.
-
+        """ Return the dimension of the distribution based on the geometry.
+    
         If the dimension can not be inferred, None is returned.
 
-        Subclassing distributions can choose to overwrite this property if different behavior is desired.
         """
+        return self.geometry.par_dim
 
+    def _infer_dim_of_mutable_variables(self):
+        """ Infer the dimension of the mutable variables of the distribution. """
         # Get all mutable variables
         mutable_vars = self.get_mutable_variables()
 
+        # Check if mutable_vars is empty, no dimension can be inferred
+        if not mutable_vars:
+            return None
+
         # Loop over mutable variables and get range dimension of each and get the maximum
         max_len = max([infer_len(getattr(self, var)) for var in mutable_vars])
 
         return max_len if max_len > 0 else None
 
     @property
     def geometry(self):
-        if self.dim != self._geometry.par_dim:
-            if isinstance(self._geometry,_DefaultGeometry):
-                self.geometry = self.dim
-            else:
-                raise Exception("Distribution Geometry attribute is not consistent with the distribution dimension ('dim')")
+        """ Return the geometry of the distribution.
+
+        The geometry can be automatically inferred from the mutable variables of the distribution.
+
+        """ 
+
+        inferred_dim = self._infer_dim_of_mutable_variables()
+
+        # If inconsistent geometry dimensions, raise an exception
+        if inferred_dim and self._geometry.par_dim and inferred_dim > 1 and self._geometry.par_dim != inferred_dim:
```

Comment:
Just wanted to ask about the `inferred_dim` > 1 condition, is there a reason to exclude raising this exception when inferred_dim ==1?.

Reply:
Updated to improve readability